src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -71,13 +71,13 @@ create_topic_properties_update(
      * configuration in topic table, the only difference is the replication
      * factor, if not set in the request explicitly it will not be overriden.
      */
-    update.properties.compaction_strategy.op = op_t::set;
-    update.properties.compression.op = op_t::set;
-    update.properties.segment_size.op = op_t::set;
-    update.properties.timestamp_type.op = op_t::set;
-    update.properties.retention_bytes.op = op_t::set;
-    update.properties.retention_duration.op = op_t::set;
-    update.properties.shadow_indexing.op = op_t::set;
+    update.properties.compaction_strategy.op = op_t::remove;
+    update.properties.compression.op = op_t::remove;
+    update.properties.segment_size.op = op_t::remove;
+    update.properties.timestamp_type.op = op_t::remove;
+    update.properties.retention_bytes.op = op_t::remove;
+    update.properties.retention_duration.op = op_t::remove;
+    update.properties.shadow_indexing.op = op_t::remove;
```

Comment:
hmm, i think this is literally the correct behavior, but i'm less certain it's what we actually want.